internal/app/init/pkg/system/health/health_test.go

```diff
@@ -127,7 +127,7 @@ func (suite *CheckSuite) TestCheckAbort() {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
-		case <-time.After(10 * time.Millisecond):
+		case <-time.After(50 * time.Millisecond):
```

Comment:
Unsure if it'd complicate it more, but would it work to close a chan here and wait for it/block on L144?

Reply:
this code actually shouldn't run as it should be aborted on context, as healthcheck timeout is 1ms, so it should execute the branch above, `ctx.Done()` (`ctx` passed here is different from `ctx` below in this function, it is setup with `Timeout` deadline)

Reply:
so in fact looks like in the failure, health is okay, so it means that healthcheck has run (otherwise state would be undefined/nil), but it exited on this branch `time.After(10ms)`, while it should never take it